example_copy_test.go

```diff
@@ -215,6 +216,37 @@ func ExampleCopy_remoteToRemote() {
 	// sha256:7cbb44b44e8ede5a89cf193db3f5f2fd019d89697e6b87e8ed2589e60649b0d1
 }
 
+func ExampleCopy_remoteToRemoteWithMount() {
+	reg, err := remote.NewRegistry(remoteHost)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	src, err := reg.Repository(ctx, "source")
+	if err != nil {
+		panic(err) // Handle error
+	}
+	dst, err := reg.Repository(ctx, "target")
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tagName := "latest"
+
+	opts := oras.CopyOptions{}
```

Comment:
What if someone reuse `oras.CopyOptions`?

Reply:
The CopyOptions after a WithMount would only be usable when the same source and destination are not changed.  That is not ideal.  A shallow copy of the `oras.CopyOptions` before calling `WithMount()` solves this problem.  That is how I use this in my code.

Reply:
It would be ideal if we would reuse the CopyOptions. Currently, only PackOptions is not re-usable.

Reply:
We could change the signature of the `WithMount()` to return a modified copy of the receiver.  It would not modify the receiver so `opts` below could be reused.  It would then be used like so:
```go
desc, err := oras.Copy(ctx, src, tagName, dst, tagName, opts.WithMount("source", dst.(registry.Mounter), nil))
```
I think that would reduce the misuse and make it more obvious what it does.